backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/IOSGraphics.java

```diff
@@ -603,6 +603,16 @@ public void setResizable(boolean resizable) {
 	public void setVSync (boolean vsync) {
 	}
 
+	/** Sets the preferred framerate for the application. Use 0 to target the maximum refresh rate of the display.
+	 * Default is 60.
+	 *
+	 * @param fps the preferred fps */
+	@Override
+	public void setForegroundFPS (int fps) {
+		config.preferredFramesPerSecond = fps;
+		viewController.setPreferredFramesPerSecond(config.preferredFramesPerSecond);

```

Comment:
@crykn I think the relevant iOS class in our case is the `GLKViewController` in which there doesn't seem to be a value to target device refresh rate (https://developer.apple.com/documentation/glkit/glkviewcontroller/1620723-framespersecond). I think we need to remove the java docs about 0 as maximum refresh rate.

The way I understand the preferred fps configuration on mobile platforms is as Max fps. It is useful to set the target ideal fps you want the game to run on and to hypothetically limit it if there was a device with higher fps (doesn't exist on iOS afaik yet). On the other hand (again, at least on mobile) IMO, an API tn set the FPS programatically may make sense if you want to temporarily on a specific screen or for dramatic effect limit the fps to less but this is something that you could easily achieve by clamping the `delta` in the project (until/if we make a delta clamping API on libGDX). For these use cases, I'm not sure it makes sense to overwite the preferred fps set on config.

Reply:
Thinking forward, on Android 30 there's a way to set preferred fps https://developer.android.com/guide/topics/media/frame-rate. On the section **Usage for games or other non-video apps** it also seems to recommend using it as a limit and setting it once. I think on mobile platforms we should only support setting fps through configuration and not implement the setForegroundFps() on `Graphics`. For playing with fps we should recommend clamping the delta.

Reply:
Yes, you're right, it seems like I should have taken a closer look. I updated the javadoc accordingly.

About your second point:
> I think on mobile platforms we should only support setting fps through configuration and not implement the setForegroundFps() on Graphics

I don't think we should just not support setting the fps on mobile platforms. I suppose it would be better to implement it and offer the possibility, but maybe discourage using it in our documentation.

Reply:
Well, it's not a big deal, my concern is that it creates more confusion and bad practices for no benefit.

My other point was, why do we do this? 

`config.preferredFramesPerSecond = fps;`

The `IOSApplicationConfiguration` `config` object is passed by the project, I don't think we should mutate it.

Reply:
> My other point was, why do we do this?
> `config.preferredFramesPerSecond = fps;`

I did this to mirror both desktop backends. They use the config object to save their current configuration even if later changed, see `setUndecorated()`, `setResizable()`, etc. (though, apparently not in `LwjglGraphics#setVsync()`, maybe that should get changed).

Reply:
Ok, that's pretty bad. If we want this we should probably have a variable that gets set initially by config and used from then onwards and, in that case, the same way we have a setter we should probably have a getter.